From 6ac980c05d11105a75928c600768c5f8cecbfec8 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Sat, 27 Oct 2012 18:09:04 +0200
Subject: [PATCH] don't throw exceptions

* maybe not best way to exit, but, palm qmake config disables them explicitly
  QMAKE_CXXFLAGS          += $$QMAKE_CFLAGS -fno-rtti -fno-exceptions
  ./usr/lib/qmake-webos/mkspecs/common/g++.conf:QMAKE_CXXFLAGS            += 11838QMAKE_CFLAGS -fno-rtti -fno-exceptions
  ./usr/lib/qmake-webos/mkspecs/common/g++.conf:QMAKE_LFLAGS              += -fno-rtti -fno-exceptions

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 libconnman-qt/networkservice.cpp    | 2 +-
 libconnman-qt/networktechnology.cpp | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libconnman-qt/networkservice.cpp b/libconnman-qt/networkservice.cpp
index d90bbf0..ea25d2f 100644
--- a/libconnman-qt/networkservice.cpp
+++ b/libconnman-qt/networkservice.cpp
@@ -35,7 +35,7 @@ NetworkService::NetworkService(const QString &path, const QVariantMap &propertie
 
     if (!m_service->isValid()) {
         qDebug() << "invalid service: " << path;
-        throw -1; // FIXME
+        exit(1); // FIXME
     }
 
     m_propertiesCache = properties;
diff --git a/libconnman-qt/networktechnology.cpp b/libconnman-qt/networktechnology.cpp
index dc52660..483e38a 100644
--- a/libconnman-qt/networktechnology.cpp
+++ b/libconnman-qt/networktechnology.cpp
@@ -24,7 +24,7 @@ NetworkTechnology::NetworkTechnology(const QString &path, const QVariantMap &pro
 
     if (!m_technology->isValid()) {
         qDebug() << "invalid technology: " << path;
-        throw -1; // FIXME
+        exit(1); // FIXME
     }
 
     m_propertiesCache = properties;
-- 
1.7.12.4

